File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        return_list = []
        i=0
        j= len(nums) - 1

        while len(return_list) != len(nums):
            total = 1
            count = 0
            while count != len(nums): 
                
                if i != j:
                    total *= nums[j]
                else:
                    i += 1
                
                if j == 0:
                    j = len(nums)
                
                j-=1

                count = count + 1

            return_list.append(total)

        return return_list

File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {

        vector<int> dp(nums.size(),0);
        dp[0] = 1;

        for (int i = 0; i < nums.size();i++){
            if (dp[i] == 1) {
                for (int k = 1; k<nums[i]+1;k++) {
                    if (i+k < nums.size()) {
                        dp[i+k] = 1;
                    }
                }
            }
        } 

        if (dp[nums.size()-1] == 1) {
            return true;
        }    
        return false;
    }
};
